src/Language/Pirouette/Example/Syntax.hs

```diff
@@ -318,7 +317,7 @@ ident = label "identifier" $ do
 
 typeName :: Parser String
 typeName = label "type-identifier" $ do
-  t <- lexeme ((:) <$> upperChar <*> many (alphaNumChar <|> char '_'))
+  t <- lexeme ((:) <$> upperChar <*> many (alphaNumChar <|> char '_' <|> char '@'))
```

Comment:
⚠️ WARNING: extra nitpickiness approaching.

I've been bitten in the past by trying to make identifiers too similar to full expressions. In this case, `f @Bool` is turned into `f@Bool`, and those are just too similar. May I suggest changing to a different symbol, so that we can more easily see in our code that something was a monomorphized function as opposed to a type application?

Reply:
We relied on `'@'` because it was what was used by monomorphization, so we needed to be able to type those names; I changed it to `monoNameSep` and defined that to be `'!'` instead.